Amicabilibus+RemoveDuplicates/src/HarryPotterReader.java

```diff
@@ -0,0 +1,118 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class HarryPotterReader {
+	
+	public static void main(String[] args) {
+		
+		Map<String, Integer> counted = countDuplicates(readFile("harry_potter.txt"));
+		LinkedHashMap<String, Integer> sorted = sort(counted);
+		
+		for(String key: sorted.keySet()) {
+			System.out.println(key + " : " + sorted.get(key));
+		}
+	}
+	
+	public static ArrayList<String> readFile(String fileName) {
+		
+		ArrayList<String> keys= new ArrayList<String>();
+		
+		try {
+			
+			Scanner s = new Scanner(new File(fileName));
+			System.out.println("File is open");
+			
+			while(s.hasNext()) {
+				
+				String newWord = s.next();
+				newWord = trimIt(newWord);
+				
+				if(newWord != "") {
+					keys.add(newWord);
+				}
+			}
+			s.close();
+			
+		}catch(Exception e) {
+			
+			System.out.println("Problem...");
+		}
+		
+		return keys;
+		
+	}
+	
+	public static HashMap<String, Integer> countDuplicates(ArrayList<String> keys){
+		
+		HashMap<String, Integer> dictionary = new HashMap<String, Integer>();
+		
+		for(String key:keys) {
+			if(dictionary.containsKey(key)) {
+				dictionary.put(key, dictionary.get(key) +1);
+			}else {
+				dictionary.put(key, 1);
+			}
+			
+		}
+		
+		return dictionary;
+		
+	}
+	
+	
+	public static String trimIt(String word) {
+		
+		word = word.toLowerCase();
+		
+		int firstIndex = 0;
+		int lastIndex = word.length()-1;
+		
+		boolean isWord = false;
+		for(char ch : word.toCharArray()) {
+			if(Character.valueOf(ch)>=97 && Character.valueOf(ch)<=123) {
+				isWord = true;
```

Comment:
bigger flaw here. This code means that if one of the characters inside this string is a character between 'a' and 'z' then `isWord == true` I see what you want to do and some very small fix can be done to fix it.

Reply:
i don't quit understand what you mean... I just wanted to check if there is a character in the String because or else the rest of the code will result in an error